apps/prairielearn/elements/pl-number-input/pl-number-input.py

```diff
@@ -97,6 +97,14 @@ def format_true_ans(
             element, "comparison", ComparisonType, COMPARISON_DEFAULT
         )
 
+        # Correct answers may be specified as strings, so we need to convert them
+        # to numbers for formatting. See the note in `grade()` for why we cast to
+        # these specific types.
+        if np.iscomplexobj(correct_answer):
+            correct_answer = np.complex128(correct_answer)
```

Comment:
This may not be strictly necessary, but it keeps symmetry with what we're doing in `grade()`.